src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -870,7 +872,19 @@ private void ScanSyntaxToken(ref TokenInfo info)
                         TextWindow.AdvanceChar();
                     }
 
-                    info.Text = TextWindow.GetText(intern: true);
+                    if (_errorCount++ > 100)
```

Comment:
:bulb: I would be much more comfortable with this solution if the following changes were made:

1. Raise the threshold *significantly*, e.g. to 50000 (this number is plenty low enough to address the original issue).
1. Emit a warning indicating that the rest of the input is being ignored due to a large number of lexer errors. Perhaps the message could state that the input appears to not be C# code?